test/fika/parser_test.exs

```diff
@@ -501,6 +501,49 @@ defmodule Fika.ParserTest do
                  ]}}
              ]
     end
+
+    test "with union types" do
+      code = """
+      fn foo(a: Int | Float, b: Nothing) : :ok | {:error, String} do
+        if true do
+          :ok
+        else
+          {:error, "not ok"}
+        end
+      end
+      """
+
+      arg_union = MapSet.new([:Int, :Float])
+
+      return_union =
+        MapSet.new([
+          :ok,
+          %T.Tuple{elements: [:error, :String]}
+        ])
+
+      assert {:ok,
+              [
+                {:function, [position: {7, 120, 123}],
+                 {:foo,
+                  [
+                    {{:identifier, {1, 0, 8}, :a},
+                     {:type, {1, 0, 21}, %T.Union{types: ^arg_union}}},
+                    {{:identifier, {1, 0, 24}, :b}, {:type, {1, 0, 33}, :Nothing}}
+                  ],
+                  {:type, {1, 0, 59},
+                   %T.Union{
+                     types: ^return_union
+                   }},
+                  [
+                    {{:if, {6, 114, 119}}, {:boolean, {2, 63, 72}, true},
+                     [{:atom, {3, 76, 83}, :ok}],
+                     [
+                       {:tuple, {5, 91, 113},
+                        [{:atom, {5, 91, 102}, :error}, {:string, {5, 91, 112}, ["not ok"]}]}
+                     ]}
+                  ]}}
+              ], "\n", %{}, {7, 120}, 123} = TestParser.function_def(code)
```

Comment:
We can use the new `function_def!` function to return the function def AST node directly.